app/components/assets/booking-actions-dropdown.tsx

```diff
@@ -0,0 +1,61 @@
+import { useLoaderData } from "@remix-run/react";
+import type { loader } from "~/routes/_layout+/assets.$assetId";
+import type { CustomLink } from "../shared/actions-drop-down";
+import { ActionsDropDown } from "../shared/actions-drop-down";
+import { Button } from "../shared/button";
+
+export default function BookingActionsDropDown() {
+  const { asset } = useLoaderData<typeof loader>();
+  const assetIsCheckedOut = asset.status === "CHECKED_OUT";
+  const assetIsPartOfUnavailableKit = Boolean(
+    asset.kit && asset.kit.status !== "AVAILABLE"
+  );
+
+  const reason = asset.kit
+    ? {
+        reason: (
+          <>
+            Cannot book this asset directly because it's part of a kit. Please
+            book the{" "}
+            <Button to={`/kits/${asset.kit.id}`} target="_blank" variant="link">
+              kit
+            </Button>{" "}
+            instead.
+          </>
+        ),
+      }
+    : false;
+  const disabled =
+    assetIsCheckedOut || assetIsPartOfUnavailableKit || asset.kit;
+  const links = [
+    {
+      indexType: "asset",
+      id: asset.id,
+      disabled,
+      label: "Create new booking",
+      icon: "bookings",
+      disabledReason: reason,
+      to: `/bookings/new?assetId=${asset.id}`,
+    },
+    {
+      indexType: "asset",
+      id: asset.id,
+      label: "Add to existing booking",
+      icon: "booking-exist",
+      disabled,
+      disabledReason: reason,
+      to: `/bookings/update-existing?indexType=assets&id=${asset.id}`,
+    },
+  ] as CustomLink[];
+
+  return (
+    <div className="actions-dropdown flex">
```

Comment:
Although this works, that is now what we wanted to achieve. Please look at the example we gave.
It is not a singular dropdown, its a button group where the left item is the main button and the right item is a dropdown trigger with the different options. Let me know if you need further examples.